### ISSUE TITLE
Eigenes Ldap setup

### ISSUE THREAD

**Issue by therealr5** [fsr/fruitbasket#72]
Portunus sucks

**Comment by revol-xut** [comment]
Is there some nicer ldap frontend ?

**Comment by therealr5** [comment]
> Is there some nicer ldap frontend ?

Not really. I'd personally just open the ldaps port to the outside and use some remote LDAP management client like https://github.com/kallisti5/gqclient or similar

**Comment by revol-xut** [comment]
This looks like vomit to be frank. What is actually the problem with portunus the looks or is the some key functionality missing ?

**Comment by FugiMuffi** [comment]
> This looks like vomit to be frank. What is actually the problem with portunus the looks or is the some key functionality missing ?

I actually really like the web ui, but Portunus does have some major issues and is indeed missing important functionality, not only for our use case.

I think its general approach is far from ideal: As I understand it, Portunus sees its own database as the single source of truth, while the puppeteered OpenLDAP is merely supposed to serve a read-only copy of the data. Instead of relying on OpenLDAP as the backend, Portunus tries to reinvent the wheel and implement authentication and so on itself, just for its web ui. This is IMHO a lot of unnecessary effort, and it causes a list of problems:

- it is by design impossible to modify anything from the LDAP side, e.g. let users change their own password using `passwd`
- sha256 password hashes are the only ones it knows how to handle, so many of our migrated accounts can not even log in to the web ui (they can still use our services, because OpenLDAP can handle it just fine)
  - this is obviously bad, as there is no way for them to change their password without asking an admin for help
  - there is also no way to migrate to a different hashing algorithm, should it be necessary
- Portunus only lets you use either insecure LDAP or LDAPS (patched by us)
  - having both is useful, since dex pretty much requires LDAPS, while it is unnecessarily complicated to configure LDAPS in a sensible way for some services, which are running on the same host anyway
- having e.g. an umlaut in your name (very common) silently breaks things, because the gecos field is not being ascii-escaped. Portunus happily accepts it, but OpenLDAP does not. (patched by us)
- Portunus requires first and last names (patched by us)
  - not all existing accounts fit that requirement
  - not all humans fit that requirement, see [this](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)

I believe that using plain OpenLDAP as a base, with some (simple to use) client / management software for administration and possibly self service, would be a better approach that does not have those problems.

**Comment by therealr5** [comment]
We could try to spin up https://github.com/dnknth/ldap-ui as web ui. This looks quite promising and uses openldap as backend

**Comment by mhaehnel** [comment]
Why not use [lam](https://www.ldap-account-manager.org/) as we did on Kaki, if you just want a nice frontend for (Open-)LDAP? I actually like it and use it privately as well. Plus it has a long maintenance history, so it is not likely to go away anytime soon.
